### PR TITLE
feat: programmatic netlify.toml support & copying

### LINKED ISSUE #4: Programmatically create _redirects file through JS objects

### PR DESCRIPTION
**Pull request by andre-brdoch**
- copy netlify.toml from src to dist
- alternatively create netlify.toml programmatically

It would probably make sense to change the nuxt hook to `generate:distRemoved`, since certain things like env var might need to be accessible earlier on already. I just was not sure if there would be any downsides to this.

This is my first time contributing to an open source project, and first time using conventional commit messages, so I hope I did things correctly :) feedback is very welcome

Resolves #4

### CODE REVIEW

File: lib/module.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+const { copyFileSync, existsSync, writeFileSync } = require('fs')
+const tomlStringify = require('@iarna/toml/stringify')
 const logger = require('./logger')
 
 const HEADERS_FILE_NAME = '_headers'
 const REDIRECTS_FILE_NAME = '_redirects'
-const FILE_NAMES = [HEADERS_FILE_NAME, REDIRECTS_FILE_NAME]
+const TOML_FILE_NAME = 'netlify.toml'
+const FILE_NAMES = [HEADERS_FILE_NAME, REDIRECTS_FILE_NAME, TOML_FILE_NAME]
+let filesToCopy = FILE_NAMES.slice()
```
Comment (manniL) [inline lib/module.js:10]: What's the reason to introduce this extra variable here?

Comment (manniL) [reply lib/module.js:10]: Ah, nvm.!

File: lib/module.js [hunk 2]
```diff
@@ -19,25 +23,40 @@ module.exports = function (moduleOptions) {
...
+function programmaticallyCreateToml ({ netlifyToml }) {
+  let tomlObj = netlifyToml
+  if (typeof tomlObj === 'function') { tomlObj = netlifyToml() }
+  if (typeof tomlObj !== 'object') { throw new TypeError('"netlifyToml" must be an object, or a function that returns an object') }
+  const toml = tomlStringify(tomlObj)
+  const destination = resolve(this.options.rootDir, this.options.generate.dir, TOML_FILE_NAME)
+  writeFileSync(destination, toml)
 }
 
 function copyExistingNetlifyFiles ({ existingFilesDirectory }) {
```
Comment (manniL) [inline lib/module.js:48]: Maybe we could pass the file names to copy as an argument?

Comment (andre-brdoch) [reply lib/module.js:48]: sure, will do in the next days 👍

**Comment by manniL** [comment]
Well done! 👍🏻

**Comment by andre-brdoch** [comment]
thank you! ☺️ thanks for reviewing

**Comment by manniL** [comment]
@andre-brdoch Will bundle a release in the coming days 😉

**Pull request merged**